packages/plugin-react-navigation/test/navigation-tracker.test.ts

```diff
@@ -0,0 +1,266 @@
+import type { AppState } from '@bugsnag/core-performance'
+import { MockReactNativeSpanFactory } from '@bugsnag/js-performance-test-utilities'
+import type { ReactNativeSpanFactory } from '@bugsnag/react-native-performance'
+import { NavigationTracker } from '../lib/navigation-tracker'
+import type { NavigationContainerRef } from '@react-navigation/native'
+
+beforeEach(() => {
+  jest.useFakeTimers()
+})
+
+afterEach(() => {
+  jest.useRealTimers()
+})
+
+let appState: AppState = 'starting'
+
+const mockNavigationContainerRef = () => {
+  const unsafeActionListeners: Array<(event: any) => void> = []
+  const stateChangeListeners: Array<() => void> = []
+
+  let currentRoute: { name: string } = { name: 'initial-route' }
+
+  return {
+    addListener: jest.fn((eventName: string, listener: () => void) => {
+      if (eventName === 'state') {
+        stateChangeListeners.push(listener)
+      } else if (eventName === '__unsafe_action__') {
+        unsafeActionListeners.push(listener)
+      }
+    }),
+    changeRoute: jest.fn((routename: string) => {
+      currentRoute = { name: routename }
+    }),
+    getCurrentRoute: jest.fn(() => currentRoute),
+    isReady: jest.fn(() => true),
+    triggerStateChange: () => {
+      stateChangeListeners.forEach((listener) => { listener() })
+    },
+    triggerUnsafeAction: (noop = false) => {
+      unsafeActionListeners.forEach((listener) => { listener({ data: { action: { type: 'NAVIGATION' }, noop } }) })
+    }
+  }
+}
+
+describe('NavigationTracker', () => {
+  it('Creates a navigation span when the currentRoute changes', () => {
+    const setAppState = jest.fn((state: AppState) => { appState = state })
+    const spanFactory = new MockReactNativeSpanFactory()
+
+    const mockNavigationContainer = mockNavigationContainerRef()
+    const navigationTracker = new NavigationTracker(spanFactory as unknown as ReactNativeSpanFactory, setAppState)
+    navigationTracker.configure(mockNavigationContainer as unknown as NavigationContainerRef<ReactNavigation.RootParamList>)
+
+    expect(appState).toBe('starting')
+
+    // TODO: Initial route should not create a span??
```

Comment:
I'm guessing this got copied over from older code? worth noting it - I don't think the first route _should_ count as a navigation, but just raising my hand about this one